src/Forms/Shared/Samples/Location/IndoorPositioning/IndoorPositioning.xaml.cs

```diff
@@ -0,0 +1,204 @@
+                    WhereClause = "1=1",
+                };
+                queryParameters.OrderByFields.Add(new OrderBy(dateCreatedFieldName.Name, SortOrder.Descending));
+
+                FeatureQueryResult queryResult = await positioningTable.QueryFeaturesAsync(queryParameters);
+                Guid globalID = (Guid)queryResult.First().Attributes["GlobalID"];
+
+                // Create the indoor location data source using the tables and Guid.
+                _indoorsLocationDataSource = new IndoorsLocationDataSource(positioningTable, pathwaysTable, globalID);
+
+                MyMapView.LocationDisplay.AutoPanMode = LocationDisplayAutoPanMode.Navigation;
+                MyMapView.LocationDisplay.DataSource = _indoorsLocationDataSource;
+                _indoorsLocationDataSource.LocationChanged += LocationDisplay_LocationChanged;
+
+                await MyMapView.LocationDisplay.DataSource.StartAsync();
+            }
+            catch (Exception ex)
+            {
+                await Application.Current.MainPage.DisplayAlert(ex.GetType().Name, ex.Message, "OK");
+            }
+        }
+
+        private void LocationDisplay_LocationChanged(object sender, Location loc)
+        {
+            // Get the properties from the new location.
+            IReadOnlyDictionary<string, object> locationProperties = loc.AdditionalSourceProperties;
+
+            locationProperties.TryGetValue(LocationSourcePropertyKeys.Floor, out object floor);
+            locationProperties.TryGetValue(LocationSourcePropertyKeys.PositionSource, out object positionSource);
+            locationProperties.TryGetValue(LocationSourcePropertyKeys.SatelliteCount, out object satCount);
+            locationProperties.TryGetValue("transmitterCount", out object transmitterCount);
+
+            int newFloor = int.Parse(floor.ToString());
+
+            // Check if the new location is on a different floor.
+            if (_currentFloor == null || _currentFloor != newFloor)
+            {
+                _currentFloor = newFloor;
+
+                foreach (Layer layer in MyMapView.Map.OperationalLayers)
+                {
+                    if (_layerNames.Contains(layer?.Name) && layer is DimensionLayer dimLayer)
+                    {
+                        // Set the layer definition expression to only show data for the current floor.
+                        dimLayer.DefinitionExpression = $"VERTICAL_ORDER = {_currentFloor}";
+                    }
+                }
+            }
+
+            // Set text for satellites or beacons.
+            string countText = string.Empty;
+            if (positionSource.Equals("GNSS"))
+            {
+                countText = $"Satellite count: {satCount}";
+            }
+            else if (positionSource.Equals("BLE") || positionSource.Equals("AppleIPS"))
+            {
+                countText = $"Beacon count: {transmitterCount}";
+            }
```

Comment:
Rather than relying on the value of position source, you could just rely on the presence of "satCount" and "transmitterCount"? It would also be more future proof, and be able to display a combined set of values (a location could be based on both satellites and beacons)
Ie
```cs
if(floor != null) countText += $"Floor : {floor}\n";
if(satCount != null) countText += $"Satellite count: {satCount}\n";
if(transmitterCount != null) countText += $"Beacon count: {transmitterCount}\n";
```